ykbuild/build.rs

```diff
@@ -50,15 +40,44 @@ fn main() {
     ykllvm_dir.push("ykllvm");
     create_dir_all(&ykllvm_dir).unwrap();
 
+    // To avoid running cmake config/build/install, we only do anything with cmake if any of the
+    // following are true:
+    //   * ykllvm isn't built.
+    //   * The ykllvm submodule has changed since we last built it.
+    //   * The ykllvm directory is dirty (i.e. there are new/changed files).
+    //   * The ykllvm directory was changed from dirty to clean since we last built it.
```

Comment:
Why do we run `cmake` when the directory was changed from dirty to clean. Doesn't clean mean there's nothing to do?

Reply:
"Clean" here is the git concept of "clean" i.e. `git status` shows no changes. But if there were previously changes and now there aren't, it means that we need to rebuild to make sure that we don't have a build with the old (dirty) stuff in it.